src/js/tabs/osd.js

```diff
@@ -1487,8 +1501,12 @@ OSD.msp = {
                 result.push16(warningFlags);
                 if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
                     result.push32(warningFlags);
-
+                    // NOTE: betaflight MSP does not expect result.push8(OSD.data.osd_profiles.number);
                     result.push8(OSD.data.osd_profiles.selected + 1);
```

Comment:
I don't understand this comment added. It does expect the number of current OSD profile here: https://github.com/betaflight/betaflight/blob/eb7e6c0b39695966bffc3849debe9ddcb700781d/src/main/msp/msp.c#L2758

Reply:
MSP.c only expects `OSD.data.osd_profiles.selected` to be sent back from the Configurator, not `OSD.data.osd_profiles.number` which is implemented in Betaflight as the constant `OSD_PROFILE_COUNT`. I had a off-by-one byte alignment issue if I sent it, hence I added a comment to explain.

Anyway, this PR will need to wait until Betaflight 4.2 as 4.1 will have this feature as cli only.

Reply:
@ianrmurphy: It will still need to have `RSSI` and `LINK_QUALITY` added to `ALL_WARNINGS`.